service/controller/chart/resource/release/resource.go

```diff
@@ -135,10 +135,66 @@ func (r *Resource) findHelmV2ConfigMaps(ctx context.Context, releaseName string)
 	return len(cms.Items) > 0, nil
 }
 
-// patchAnnotations updates the chart CR annotations if they have changed.
+func (r *Resource) addAnnotation(ctx context.Context, cr *v1alpha1.Chart, key, value string) error {
```

Comment:
create `addAnnotation` function in resource struct so we can use it in multiple places.